### PR TITLE
easyeffects: add module

### PR DESCRIPTION
**Pull request by fufexan**
### Description

Added `services.easyeffects` as a PipeWire replacement for `services.pulseeffects`.

### Checklist

- [ ] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [x] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [x] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/services/pulseeffects.nix [hunk 1]
```diff
@@ -9,19 +9,16 @@ let
   presetOpts = optionalString (cfg.preset != "") "--load-preset ${cfg.preset}";
 
 in {
+  imports = [
+    (mkRemovedOptionModule [ "services" "pulseeffects" "package" ]
+      "The PipeWire equivalent of Pulseeffects is called Easyeffects.")
+  ];
```
Comment (berbiche) [inline modules/services/pulseeffects.nix:15]: We don't need to remove the option, only to set it to `pkgs.pulseeffects-legacy` and to remove the example.

**Review by berbiche — CHANGES_REQUESTED** [review]
Thanks!

File: modules/services/pulseeffects.nix (hunk 1 above)
Comment (fufexan) [reply modules/services/pulseeffects.nix:15]: I figured it should be removed since now `pkgs.pipewire-pw` is an alias to `pkgs.easyeffects`, and the `pulseeffects` binary is no longer in the package - so the service would fail.

Comment (berbiche) [reply modules/services/pulseeffects.nix:15]: Yes but people that have not switched to pipewire might still the need the option.

This is the pulseeffects service afterall.

Comment (fufexan) [reply modules/services/pulseeffects.nix:15]: Ah, that makes sense, since they would want custom packages probably. It hadn't occured to me before. Sorry for being dense :sweat_smile:

File: modules/services/easyeffects.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  };
+
+  config = mkIf cfg.enable {
+    # running easyeffects will just attach itself to gapplication service
+    # at-spi2-core is to minimize journalctl noise of:
+    # "AT-SPI: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files"
+    home.packages = with pkgs; [ easyeffects at-spi2-core ];
+
+    # Will need to add `services.dbus.packages = with pkgs; [ gnome.dconf ];`
+    # to /etc/nixos/configuration.nix for daemon to work correctly
```
Comment (sumnerevans) [inline modules/services/easyeffects.nix:34]: This feels like documentation that should be in the option, not in the source. I think we should add it to the `enable` option. I know that `pulseeffects` doesn't do this, but let's at least do it correctly for this one.

File: modules/services/easyeffects.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    systemd.user.services.easyeffects = {
+      Unit = {
+        Description = "Easyeffects daemon";
+        Requires = [ "dbus.service" ];
+        After = [ "graphical-session-pre.target" ];
+        PartOf = [ "graphical-session.target" "pipewire.service" ];
+      };
+
+      Install = { WantedBy = [ "graphical-session.target" ]; };
```
Comment (sumnerevans) [inline modules/services/easyeffects.nix:44]: nit:

```suggestion
      Install.WantedBy = [ "graphical-session.target" ];
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]
Just a couple small things. Looks good otherwise! Thanks for the contribution.

File: modules/modules.nix [hunk 4]
```diff
@@ -194,6 +194,7 @@ let
     ./services/playerctld.nix
     ./services/polybar.nix
     ./services/poweralertd.nix
+    ./services/easyeffects.nix
```
Comment (berbiche) [inline modules/modules.nix:197]: Can you keep the list ordered? Thanks

**Review by berbiche — APPROVED** [review]

File: modules/services/easyeffects.nix [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  presetOpts = optionalString (cfg.preset != "") "--load-preset ${cfg.preset}";
+
+in {
+  meta.maintainers = [ maintainers.fufexan ];
+
+  options.services.easyeffects = {
+    enable = mkEnableOption ''
+      Easyeffects daemon.
+      Add `services.dbus.packages = with pkgs; [ gnome.dconf ];`
+      to /etc/nixos/configuration.nix for daemon to work correctly
```
Comment (sumnerevans) [inline modules/services/easyeffects.nix:18]: How does this get rendered? I think it needs to be something like:

```suggestion
      Easyeffects daemon.
      Add <literal>services.dbus.packages = with pkgs; [ gnome.dconf ];</literal>
      to <filename>/etc/nixos/configuration.nix</filename> for daemon to work correctly
```

**Review by sumnerevans — CHANGES_REQUESTED** [review]

File: modules/services/easyeffects.nix (hunk 5 above)
Comment (berbiche) [reply modules/services/easyeffects.nix:18]: Right, an oversight on my end again.

The definition should use `<code>` instead of `<literal>`.
`<option>` cannot be used because this is an external option.

Comment (rycee) [reply modules/services/easyeffects.nix:18]: I think `programlisting` is nice for longer lines of code. Perhaps something like
```xml
Note, it is necessary to add
<programlisting language="nix">
services.dbus.packages = with pkgs; [ gnome.dconf ];
};
</programlisting>
to your system configuration for the daemon to work correctly.
```

File: modules/services/pulseeffects.nix [hunk 6]
```diff
@@ -16,10 +16,9 @@ in {
 
     package = mkOption {
       type = types.package;
-      default = pkgs.pulseeffects;
```
Comment (rycee) [inline modules/services/pulseeffects.nix:19]: These changes seem unrelated to the easyeffects module? If so, please do them in a separate commit.

File: modules/services/easyeffects.nix (hunk 5 above)
Comment (fufexan) [reply modules/services/easyeffects.nix]: I wasn't sure whether to include that `};`, as I thought it was a typo. Feel free to correct me if that's not the case.

**Review by Kranzes — APPROVED** [review]
Module works, tested on my desktop.

File: modules/misc/news.nix [hunk 7]
```diff
@@ -103,7 +103,7 @@ in
       entries = mkOption {
         internal = true;
         type = types.listOf entryModule;
-        default = [];
```
Comment (sumnerevans) [inline modules/misc/news.nix:106]: What happened here? These changes shouldn't be here.

**Review by sumnerevans — CHANGES_REQUESTED** [review]

File: modules/misc/news.nix (hunk 7 above)
Comment (fufexan) [reply modules/misc/news.nix:106]: Oops, probably formatting.

**Review by sumnerevans — APPROVED** [review]
